app/serializers/api/response_serializer.rb

```diff
@@ -19,9 +19,15 @@ def questionnaire
 
     # Only calculate and send it when we specifically ask for it (e.g., for push subscriptions).
     def protocol_completion
-      return object.protocol_subscription.protocol_completion if instance_options[:with_protocol_completion]
+      return protocol_completion_without_future if instance_options[:with_protocol_completion]
 
       []
     end
+
+    private
+
+    def protocol_completion_without_future
+      object.protocol_subscription.protocol_completion.reject { |entry| entry[:future].present? }
```

Comment:
move this to protocol subscription instead? Or well, it is also just representation logic, so the serializer might be a good place.

Reply:
No because there are still places where we DO want the entire protocol completion, for example when we show a progress bar, in order to calculate how much percentage someone has completed of a diary study, we do need the number of future responses. I'll add this in a comment to clarify.

Reply:
Fixed.